File: ios/StackedWins/StackedWins/Services/AuthService.swift
import Foundation

private struct Credentials: Encodable {
    let email: String
    let password: String
}

enum AuthService {
    static func register(email: String, password: String) async throws -> AuthResponse {
        try await APIClient.shared.request(
            "POST", "auth/register",
            body: AnyEncodable(Credentials(email: email, password: password)),
            authenticated: false
        )
    }

    static func login(email: String, password: String) async throws -> AuthResponse {
        try await APIClient.shared.request(
            "POST", "auth/login",
            body: AnyEncodable(Credentials(email: email, password: password)),
            authenticated: false
        )
    }

    static func me() async throws -> User {
        try await APIClient.shared.request("GET", "auth/me")
    }
}

File: web/src/services/authService.ts
import api from './api';
import type { User } from '../types';

export interface RegisterRequest {
  email: string;
  password: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

// Why: the backend's /auth/register and /auth/login both return the full user
// record (id, email, createdAt, updatedAt). Reusing the shared `User` type
// instead of re-declaring an inline subset keeps this in lockstep with the API
// and stops the auth store from receiving a structurally incomplete user.
export interface AuthResponse {
  token: string;
  user: User;
}

export const authService = {
  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/register', data);
    if (response.data.token) {
      localStorage.setItem('token', response.data.token);
    }
    return response.data;
  },

  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await api.post<AuthResponse>('/auth/login', data);
    if (response.data.token) {
      localStorage.setItem('token', response.data.token);
    }
    return response.data;
  },

  logout() {
    localStorage.removeItem('token');
  },

  getToken(): string | null {
    return localStorage.getItem('token');
  },

  isAuthenticated(): boolean {
    return !!this.getToken();
  },
};
